components/sync15/src/request.rs

```diff
@@ -331,12 +355,15 @@ where
             serde_json::to_writer(&mut self.queued, &record).unwrap();
         }
 
+        // If all the error checks pass,
+        // update the current limits
         self.post_limits.record_added(payload_length);
         self.batch_limits.record_added(payload_length);
 
         Ok(true)
     }
 
+    // This flushes/clears certain parts of the batch based on parameters
```

Comment:
Also document what `want_commit` means, and what can be returned.

I think `Flushes the current queue` is a better description.